<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>674</width>
    <height>687</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Khrypto</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>icon.ico</normaloff>icon.ico</iconset>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <widget class="QGroupBox" name="gbSettings">
      <property name="title">
       <string>Ustawienia</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3" stretch="1,0,0">
       <item>
        <layout class="QGridLayout" name="gridLayout" columnstretch="1,2">
         <property name="sizeConstraint">
          <enum>QLayout::SetMinimumSize</enum>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Metoda</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="cbMethod">
           <item>
            <property name="text">
             <string>Szyfr Cezara</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Płot</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>XOR</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>RSA</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Huffman</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Działanie</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QComboBox" name="cbAction">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <item>
            <property name="text">
             <string>Zaszyfrowanie</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Odszyfrowanie</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Klucz</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="sbKey">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="minimum">
            <number>-65356</number>
           </property>
           <property name="maximum">
            <number>65356</number>
           </property>
           <property name="value">
            <number>3</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="gbRSA">
         <property name="title">
          <string>RSA</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="sizeConstraint">
           <enum>QLayout::SetMaximumSize</enum>
          </property>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QPushButton" name="pushButton">
              <property name="text">
               <string>Generuj klucz:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="comboBox">
              <property name="currentIndex">
               <number>1</number>
              </property>
              <item>
               <property name="text">
                <string>512 bitów</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>1024 bitów</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>2048 bitów</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>4096 bitów</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>8192 bitów</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Klucz publiczny:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnRSALoad1">
              <property name="text">
               <string>Ładuj</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnRSASave1">
              <property name="text">
               <string>Zapisz</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="plainTextEdit">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Klucz prywatny:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnRSALoad2">
              <property name="text">
               <string>Ładuj</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnRSASave2">
              <property name="text">
               <string>Zapisz</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="plainTextEdit_2">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="LInput">
          <property name="text">
           <string>Tekst wejściowy:</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="btnLoad">
          <property name="text">
           <string>Ładuj z pliku</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="tInput">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnCode">
        <property name="text">
         <string>Zaszyfruj</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="LOutput">
          <property name="text">
           <string>Tekst wyjściowy:</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="btnSave">
          <property name="text">
           <string>Zapisz do pliku</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="tOutput">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>674</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuPlik">
    <property name="title">
     <string>Aplikacja</string>
    </property>
    <addaction name="actionGeneruj_losowy_klucz"/>
    <addaction name="separator"/>
    <addaction name="actionWyj_cie"/>
   </widget>
   <addaction name="menuPlik"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionGeneruj_losowy_klucz">
   <property name="text">
    <string>Generuj losowy klucz</string>
   </property>
  </action>
  <action name="actionWyj_cie">
   <property name="text">
    <string>Wyjście</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>actionWyj_cie</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>298</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
